app-backend/batch/src/main/scala/cogMetadata/OverviewBackfill.scala

```diff
@@ -0,0 +1,78 @@
+package com.rasterfoundry.batch.cogMetadata
+
+import com.rasterfoundry.batch.Job
+import com.rasterfoundry.common.{AWSLambda, RollbarNotifier}
+import com.rasterfoundry.datamodel.ProjectLayer
+import com.rasterfoundry.database.ProjectLayerDao
+import com.rasterfoundry.database.Implicits._
+import com.rasterfoundry.database.util.RFTransactor.xaResource
+
+import cats.effect.IO
+import doobie._
+import doobie.implicits._
+import doobie.postgres.implicits._
+import fs2.Stream
+
+object OverviewBackfill extends Job with RollbarNotifier {
+  val name = "backfill-layer-overviews"
+
+  def getProjectLayerSceneCount(
+      projectLayer: ProjectLayer
+  ): Stream[ConnectionIO, (ProjectLayer, Int)] =
+    fr"select count(1) from scenes_to_layers where project_layer_id = ${projectLayer.id}"
+      .query[Int]
+      .stream map { (projectLayer, _) }
+
+  /** We know suppressing this warning is fine, because the query for project layers
+    * explicitly filters out any that don't have a project id
+    */
+  @SuppressWarnings(Array("OptionGet"))
+  def kickoffOverviewGeneration(
+      projectLayer: ProjectLayer
+  ): IO[Unit] = IO {
+    logger.info(
+      s"Kicking off lambda overview generation for layer ${projectLayer.id}")
+    if (AWSLambda.runLambda) {
+      AWSLambda
+        .kickoffLayerOverviewCreate(
+          projectLayer.projectId.get,
+          projectLayer.id,
+          "RequestResponse"
+        )
+    } else {
+      logger.info("Sleeping for 10 seconds to pretend to do work")
+      Thread.sleep(10000)
+    }
+  }
+
+  // Giant number inside listQ is because listQ needs a limit parameter, but we don't actually
+  // want to limit
+  val projectLayers: Stream[ConnectionIO, ProjectLayer] =
+    ProjectLayerDao.query
+      .filter(fr"project_id IS NOT NULL")
+      .listQ(1000000)
+      .stream
+  val projectLayersWithSceneCounts: Stream[ConnectionIO, (ProjectLayer, Int)] =
+    projectLayers.flatMap(getProjectLayerSceneCount)
+
+  def runJob(args: List[String]): IO[Unit] = {
+    xaResource
+      .use(
+        t =>
+          projectLayersWithSceneCounts
+            .filter({
+              case (_, n) => n <= 300 && n > 0
+            })
+            .transact(t)
+            .parEvalMap(20)({
```

Comment:
This appears to map onto logical threads, so requires a larger machine than I made available initially. Unclear whether we should just throw a larger machine at the problem (n CPUs = requested threads + 1) or whether I should start something with fibers, since in RequestResponse invocation each thread is going to spent a lot of time waiting. I'm learning toward fibers. :heart: fibers

Reply:
On second thought, this doesn't really make sense to invest time in -- I bumped the concurrency to 36, which happens to match the size of a c5.9xlarge. I think it should take about 7 hours to run everything, so we can fire up a c5.9xlarge (we don't need db access to run this job), pull the container, and run it for 7 hours for 7 x $1.53 = $10.71

Reply:
Or we can just limit in lambda and hope that we get everything when we get throttled at a reasonable level

Reply:
There's an alternate strategy where we set lambda reserved concurrency for the function and kick the backfill off as events. I don't think that's a good strategy.

Lambda says invocations that get kicked off once you're at the concurrency limit for a function get throttled, and that they'll retry for up to 6 hours. I don't think that will be enough time, for one, and also if this happens to have a surprising impact on db performance (adding an extra 30 requests on a constant basis for many hours should be fine, but I'm surprised all the time), I don't think it will be easy to cancel the already-invoked lambda functions.